src/app/components/client/SubscriptionCheck.tsx

```diff
@@ -4,22 +4,21 @@
 
 "use client";
 
-import { getSession, useSession } from "next-auth/react";
+import { useSession } from "next-auth/react";
 import { useEffect } from "react";
 import { hasPremium } from "../../functions/universal/user";
 
 export const SubscriptionCheck = () => {
-  const { update } = useSession();
+  const { data: session, update } = useSession();
 
   useEffect(() => {
     async function updateSession() {
-      const session = await getSession();
-      if (!hasPremium(session?.user)) {
-        await update();
-      }
+      await update();
     }
 
-    void updateSession();
+    if (!hasPremium(session?.user)) {
```

Comment:
In `<PremiumBadge>` we did not check if the user already has Premium and did the check either way. Is this something we’d like to keep doing?

Reply:
No actually now that I think about it I think the `hasPremium` check is explicitly not useful... checking this more often serves two purposes:

1. ensure that we give subscribed users access to features for which they are entitled on all devices as soon as possible
2. remove premium features as soon as cancellation has taken place